File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <vector>

using namespace std;

vector<int> solution(int n, long long left, long long right) {
    vector<int> answer;
    answer.reserve(right - left + 1); // 미리 메모리 할당

    for (long long i = left; i <= right; ++i) {
        int row = i / n;
        int col = i % n;
        answer.push_back(max(row, col) + 1);
    }

    return answer;
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int n, long left, long right) {
        
        return LongStream.range(left, right + 1)
            .mapToInt(i -> {
                int y = (int)(i / n);
                int x = (int)(i % n);
                int res = Math.max(y, x) + 1;
                return res;
            })
            .toArray();
        
    }
}